lib/mayaUsd/ufe/UsdAttribute.cpp

```diff
@@ -977,58 +584,79 @@ Ufe::UndoableCommand::Ptr UsdAttributeEnumString::setCmd(const std::string& valu
 
 Ufe::AttributeEnumString::EnumValues UsdAttributeEnumString::getEnumValues() const
 {
-    PXR_NS::TfToken tk(name());
-    auto            attrDefn = fPrim.GetPrimDefinition().GetSchemaAttributeSpec(tk);
-    if (attrDefn && attrDefn->HasAllowedTokens()) {
-        auto                         tokenArray = attrDefn->GetAllowedTokens();
-        std::vector<PXR_NS::TfToken> tokenVec(tokenArray.begin(), tokenArray.end());
-        EnumValues                   tokens = PXR_NS::TfToStringVector(tokenVec);
-        return tokens;
-    }
-#ifdef UFE_V4_FEATURES_AVAILABLE
-#if (UFE_PREVIEW_VERSION_NUM >= 4010)
-    else {
-        EnumValues                      result;
-        UsdShaderAttributeDef::ConstPtr shaderAttrDef
-            = std::dynamic_pointer_cast<const UsdShaderAttributeDef>(fAttrDef);
-        if (shaderAttrDef) {
-            const auto& shaderProperty = shaderAttrDef->shaderProperty();
-            if (shaderProperty) {
-                const auto& options = shaderProperty->GetOptions();
-                for (const auto& option : options) {
-                    result.push_back(option.first.GetString());
-                }
-            }
-        }
-        return result;
-    }
-#endif
-#endif
-    return EnumValues();
+    return UsdAttribute::getEnumValues();
 }
 
 //------------------------------------------------------------------------------
-// TypedUsdAttribute<T>:
+// UsdAttributeEnumToken:
 //------------------------------------------------------------------------------
 
-#ifdef UFE_V4_FEATURES_AVAILABLE
-template <typename T>
-TypedUsdAttribute<T>::TypedUsdAttribute(
-    const UsdSceneItem::Ptr&           item,
-    const PXR_NS::UsdPrim&             prim,
-    const Ufe::AttributeDef::ConstPtr& attrDef)
-    : Ufe::TypedAttribute<T>(item)
-    , UsdAttribute(prim, attrDef)
+UsdAttributeEnumToken::UsdAttributeEnumToken(
```

Comment:
We do not have to check what the Vt holds because we now have a completely separate class that deals exclusively with TfToken USD data.